src/pages/index.js

```diff
@@ -49,30 +131,68 @@ const enableValidation = (settings) => {
 
 enableValidation(settings);
 
-// Section class Instance
+/*-----------------------------------------------------------------*/
+/*             Class Instances                          */
+/*-----------------------------------------------------------------*/
+
+// Section instance
 const cardSection = new Section(
   {
-    items: initialCards,
+    items: [],
     renderer: (data) => {
-      cardSection.addItem(createCard(data));
+      cardSection.addItem(createCard(data), "append");
     },
   },
   cardContainer
 );
-cardSection.renderItems();
 
-// UserInfo class instance
-const userInfo = new UserInfo({
-  profileName: "#profile_name",
-  profileJob: "#profile_job",
+// Edit Avatar instance
+const popupWithAvatarForm = new PopupWithForm(
+  "#edit-avatar-modal",
+  (formData) => {
+    savingBtn.textContent = "Saving...";
+    const avatarLink = formData.avatar;
+    api
+      .changeProfilePic(avatarLink)
+      .then((data) => {
+        document.getElementById("profile__avatar").src = data.avatar;
+        popupWithAvatarForm.close();
+      })
+      .catch((err) => console.error(err));
+  }
+);
+
+popupWithAvatarForm.setEventListeners();
+
+changeAvatarbtn.addEventListener("click", () => {
+  // assign them to the value of the corresponding input elements
+  api
+    .getUserInfo()
+    .then((userData) => {
+      avatarInput.value = userData.avatar;
+      popupWithAvatarForm.open();
+    })
+    .catch((err) => console.error(err))
+    .finally(() => {
+      savingBtn.textContent = "Create";
+    });
 });
 
 // Edit modal instance
 const popupWithEditForm = new PopupWithForm(
   "#profile-edit-modal",
   (formData) => {
-    userInfo.setUserInfo(formData);
-    popupWithEditForm.close();
+    savingBtn.textContent = "Saving...";
```

Comment:
**COULD BE IMPROVED**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```